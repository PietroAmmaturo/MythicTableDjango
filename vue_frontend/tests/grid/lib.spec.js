/* eslint-disable */
import * as gridfinder from 'gridfinder';

describe('findGrid', () => {
    Object.entries({
        'perfect clicks': {
            sub: {
                '1': {
                    width: 100,
                    points: [
                        { x: 0, y: 0 },
                        { x: 100, y: 100 },
                        { x: 200, y: 200 },
                    ],
                },
                '2': {
                    width: 72,
                    points: [
                        { x: 0, y: 0 },
                        { x: 72, y: 144 },
                        { x: 288, y: 72 },
                    ],
                },
            },
        },
        'very good clicks': {
            width: 100,
            points: [
                { x: 0, y: 0 },
                { x: 110, y: 98 },
                { x: 312, y: 302 },
            ],
        },
        '60 perspective_inn': {
            bg: 'https://i.redd.it/9xy7jbjw4wv11.jpg',
            width: 60,
            sub: {
                refine_grid: {
                    points: [
                        { x: 595.977798334875, y: 428.898055242918 },
                        { x: 595.977798334875, y: 306.6018226234514 },
                        { x: 721.7187789084181, y: 306.6018226234514 },
                        { x: 721.7187789084181, y: 431.48177846727293 },
                    ],
                    sub: {
                        'half 1': {
                            points: [{ x: 657.1258094357077, y: 375.50110860624955 }],
                        },
                        'half 2': {
                            points: [{ x: 657.9870490286771, y: 429.75929631770305 }],
                        },
                    },
                },
                '1': {
                    points: [
                        { x: 1511.4754856614245, y: 1746.5968996639313 },
                        { x: 1580.3746530989824, y: 1917.9838735461417 },
                        { x: 1575.2072155411654, y: 2163.43757985986 },
                        { x: 1575.2072155411654, y: 1979.9932309306596 },
                        { x: 1637.2164662349676, y: 1919.7063556957116 },
                    ],
                },
                rectangular: {
                    shape: 'rect',
                    points: [
                        { x: 403.7346278317152, y: 174.7502656608301 },
                        { x: 463.9935275080906, y: 287.73535121740133 },
                    ],
                },
            },
        },
        '256 farmhouse': {
            bg: 'https://i.redd.it/m6l4go8l56v51.jpg',
            width: 256,
            sub: {
                min: {
                    points: [
                        { x: 2597.300613496932, y: 1813.9973492199026 },
                        { x: 5119.018404907975, y: 2562.9583759811503 },
                        { x: 5119.018404907975, y: 2830.9352571159084 },
                    ],
                },
                '1': {
                    points: [
                        // Too small.
                        { x: 5119.018404907975, y: 2830.935257115908 },
                        { x: 5112.147239263804, y: 2562.95837598115 },
                        { x: 3600.490797546012, y: 2549.215971820393 },
                    ],
                },
                bad: {
                    points: [
                        // Selects a too small grid!
                        { x: 1288.880666049953, y: 4604.322132248466 },
                        { x: 2573.617021276596, y: 1811.061000713393 },
                        { x: 3601.406105457909, y: 2557.035783615935 },
                    ],
                    sub: {
                        '+1': {
                            points: [{ x: 3601.406105457909, y: 2557.035783615935 }],
                        },
                        '+2': {
                            points: [
                                { x: 5114.079555966698, y: 1811.061000713393 },
                                { x: 4102.867715078631, y: 3332.020696964687 },
                            ],
                        },
                        '+a lot': {
                            // Lots and lots of points, good result.
                            points: [
                                { x: 5114.079555966698, y: 1811.061000713393 },
                                { x: 4102.867715078631, y: 3332.020696964687 },
                                { x: 3327.881591119334, y: 1790.3394789661 },
                                { x: 3601.406105457909, y: 3323.732088265769 },
                                { x: 2565.328399629972, y: 3323.732088265769 },
                                { x: 1284.736355226642, y: 3854.203044996466 },
                                { x: 2047.289546716003, y: 3858.347349345924 },
                                { x: 2043.145235892692, y: 4608.466436597925 },
                                { x: 2569.472710453284, y: 4608.466436597925 },
                                { x: 5114.079555966698, y: 2561.180087965393 },
                                { x: 5122.368177613321, y: 2822.271261981283 },
                                { x: 5114.079555966698, y: 3323.732088265769 },
                            ],
                        },
                    },
                },
            },
        },
        rectangular: {
            shape: 'rect',
            width: 10,
            height: 30,
            sub: {
                two: {
                    points: [
                        { x: 20, y: 30 },
                        { x: 10, y: 60 },
                    ],
                },
                three: {
                    points: [
                        { x: 20, y: 30 },
                        { x: 10, y: 60 },
                        { x: 40, y: 0 },
                    ],
                },
                offset: {
                    points: [
                        { x: 25, y: 35 },
                        { x: 15, y: 65 },
                    ],
                },
            },
        },
    }).forEach(function run([name, cfg]) {
        let { shape, bg, points = [], sub = {} } = cfg;

        if (points.length) {
            it(name, function() {
                expect(gridfinder.findGrid(points, { shape })).toMatchSnapshot();
            });
        }

        let subs = Object.entries(sub);
        describe(name, function() {
            for (let [sub_name, { points: p = [], ...rest }] of subs) {
                run([
                    sub_name,
                    {
                        ...cfg,

                        points: [...points, ...p],
                        sub: undefined,

                        ...rest,
                    },
                ]);
            }
        });

        expect(points.length || subs.length).toBeTruthy();
    });
});

describe('findOffset', () => {
    it('perfect', () => {
        expect(
            gridfinder.findOffset({
                cellSize: 99,
                points: [{ x: 0, y: 0 }],
            })).toEqual({ x: 0, y: 0 });
        expect(
            gridfinder.findOffset({
                cellSize: 99,
                points: [{ x: 10, y: 10 }],
            })).toEqual({ x: 10, y: 10 });
        expect(
            gridfinder.findOffset({
                cellSize: 100,
                points: [
                    { x: 98, y: 202 },
                    { x: 402, y: 298 },
                ],
            })).toEqual({ x: 0, y: 0 });
        expect(
            gridfinder.findOffset({
                cellSize: 100,
                points: [
                    { x: 103, y: 215 },
                    { x: 407, y: 311 },
                ],
            })).toEqual({ x: 5, y: 13 });
        expect(
            gridfinder.findOffset({
                cellSize: { width: 10, height: 20 },
                points: [
                    { x: 13, y: 23 },
                    { x: 11, y: 59 },
                ],
            })).toEqual({ x: 2, y: 1 });
    });
});
