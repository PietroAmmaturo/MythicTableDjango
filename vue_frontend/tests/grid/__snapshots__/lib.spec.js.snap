// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findGrid 60 perspective_inn 1 1`] = `
Object {
  "cellSize": Object {
    "height": 59.54866859941837,
    "width": 59.54866859941837,
  },
  "offset": Object {
    "x": 27.630823630663443,
    "y": 16.16672531810564,
  },
}
`;

exports[`findGrid 60 perspective_inn rectangular 1`] = `
Object {
  "cellSize": Object {
    "height": 112.98508555657122,
    "width": 60.25889967637539,
  },
  "offset": Object {
    "x": -18.077669902912533,
    "y": -51.21990545231233,
  },
}
`;

exports[`findGrid 60 perspective_inn refine_grid 1`] = `
Object {
  "cellSize": Object {
    "height": 125.74098057354308,
    "width": 125.74098057354308,
  },
  "offset": Object {
    "x": -32.72710453284037,
    "y": 54.04341830541574,
  },
}
`;

exports[`findGrid 60 perspective_inn refine_grid half 1 1`] = `
Object {
  "cellSize": Object {
    "height": 62.87049028677154,
    "width": 62.87049028677154,
  },
  "offset": Object {
    "x": 29.798889916743384,
    "y": -7.406024207960582,
  },
}
`;

exports[`findGrid 60 perspective_inn refine_grid half 2 1`] = `
Object {
  "cellSize": Object {
    "height": 62.87049028677154,
    "width": 62.87049028677154,
  },
  "offset": Object {
    "x": 29.971137835337277,
    "y": -9.12848472302419,
  },
}
`;

exports[`findGrid 256 farmhouse 1 1`] = `
Object {
  "cellSize": Object {
    "height": 138.04796430563303,
    "width": 138.04796430563303,
  },
  "offset": Object {
    "x": 8.953337051496268,
    "y": 70.80786793400029,
  },
}
`;

exports[`findGrid 256 farmhouse bad +1 1`] = `
Object {
  "cellSize": Object {
    "height": 8.28860869891713,
    "width": 8.28860869891713,
  },
  "offset": Object {
    "x": -4.139983052459229,
    "y": -4.144304349458196,
  },
}
`;

exports[`findGrid 256 farmhouse bad +2 1`] = `
Object {
  "cellSize": Object {
    "height": 255.01325932778298,
    "width": 255.01325932778298,
  },
  "offset": Object {
    "x": 20.997841504778442,
    "y": 17.95427024556227,
  },
}
`;

exports[`findGrid 256 farmhouse bad +a lot 1`] = `
Object {
  "cellSize": Object {
    "height": 255.84212149244524,
    "width": 255.84212149244524,
  },
  "offset": Object {
    "x": 6.207256655360515,
    "y": 6.110142328299374,
  },
}
`;

exports[`findGrid 256 farmhouse bad 1`] = `
Object {
  "cellSize": Object {
    "height": 8.28860869891713,
    "width": 8.28860869891713,
  },
  "offset": Object {
    "x": -4.140417879597938,
    "y": -4.144304349458214,
  },
}
`;

exports[`findGrid 256 farmhouse min 1`] = `
Object {
  "cellSize": Object {
    "height": 53.6535700300222,
    "width": 53.6535700300222,
  },
  "offset": Object {
    "x": 21.929252055866073,
    "y": -11.780323912012099,
  },
}
`;

exports[`findGrid perfect clicks 1 1`] = `
Object {
  "cellSize": Object {
    "height": 100,
    "width": 100,
  },
  "offset": Object {
    "x": 0,
    "y": 0,
  },
}
`;

exports[`findGrid perfect clicks 2 1`] = `
Object {
  "cellSize": Object {
    "height": 72,
    "width": 72,
  },
  "offset": Object {
    "x": 0,
    "y": 0,
  },
}
`;

exports[`findGrid rectangular offset 1`] = `
Object {
  "cellSize": Object {
    "height": 30,
    "width": 10,
  },
  "offset": Object {
    "x": -5,
    "y": 5,
  },
}
`;

exports[`findGrid rectangular three 1`] = `
Object {
  "cellSize": Object {
    "height": 30,
    "width": 10,
  },
  "offset": Object {
    "x": 0,
    "y": 0,
  },
}
`;

exports[`findGrid rectangular two 1`] = `
Object {
  "cellSize": Object {
    "height": 30,
    "width": 10,
  },
  "offset": Object {
    "x": 0,
    "y": 0,
  },
}
`;

exports[`findGrid very good clicks 1`] = `
Object {
  "cellSize": Object {
    "height": 104,
    "width": 104,
  },
  "offset": Object {
    "x": 2,
    "y": -5.333333333333333,
  },
}
`;
